packages/edit-site/src/components/sidebar-edit-mode/page-panels/page-slug.js

```diff
@@ -0,0 +1,188 @@
+/**
+ * WordPress dependencies
+ */
+import { useSelect, useDispatch } from '@wordpress/data';
+import {
+	safeDecodeURIComponent,
+	filterURLForDisplay,
+	cleanForSlug,
+} from '@wordpress/url';
+import { useState, useMemo } from '@wordpress/element';
+import { __experimentalInspectorPopoverHeader as InspectorPopoverHeader } from '@wordpress/block-editor';
+import { __ } from '@wordpress/i18n';
+import {
+	TextControl,
+	ExternalLink,
+	__experimentalHStack as HStack,
+	__experimentalVStack as VStack,
+	__experimentalText as Text,
+	Dropdown,
+	Button,
+} from '@wordpress/components';
+import { store as coreStore } from '@wordpress/core-data';
+
+export const PERMALINK_POSTNAME_REGEX = /%(?:postname|pagename)%/;
+
+function usePostPermalink( record, isEditable ) {
```

Comment:
```suggestion
function getPostPermalink( record, isEditable ) {
```
Nit: This is a "getter", so I think we should avoid using hook naming conventions.

Reply:
Right! I thought it would be a hook when started implementing the function 😄 . I renamed.